src/isomorphic/modern/class/ReactBaseClasses.js

```diff
@@ -27,6 +27,9 @@ function ReactComponent(props, context, updater) {
   this.refs = emptyObject;
   // We initialize the default updater but the real one gets injected by the
   // renderer.
+  if (this.updater && __DEV__) {
+      warning(this instanceof Constructor, 'This object has a property called updater which will be overwritten by React default updater');
```

Comment:
The code you've added won't work because it happens too early. You'll see what I mean if you write a test.